scalyr_agent/monitors_manager.py

```diff
@@ -140,12 +140,13 @@ def stop_manager(self, wait_on_join=True, join_timeout=5):
             except:
                 log.exception('Failed to stop the metric log without join due to an exception')
 
-        for monitor in self.__running_monitors:
-            # noinspection PyBroadException
-            try:
-                monitor.stop(join_timeout=1)
-            except:
-                log.exception('Failed to stop the metric log due to an exception')
+        if wait_on_join:
+            for monitor in self.__running_monitors:
+                # noinspection PyBroadException
+                try:
+                    monitor.stop(join_timeout=1)
```

Comment:
I know this isn't part of your PR, but can you change this `join_timeout` to be dependent on the `join_timeout` parameter.  Just take a snapshot of the current time at the start of the method, and then just always doing a join_timeout of `method_start_time + 5 - time.time()` with maybe checking for zero.

Reply:
@czerwingithub : I've updated the PR with what I think you were asking for.